crates/optimism/rpc/src/witness.rs

```diff
@@ -22,9 +24,13 @@ pub struct OpDebugWitnessApi<Provider, EvmConfig> {
 
 impl<Provider, EvmConfig> OpDebugWitnessApi<Provider, EvmConfig> {
     /// Creates a new instance of the `OpDebugWitnessApi`.
-    pub fn new(provider: Provider, evm_config: EvmConfig) -> Self {
+    pub fn new(
+        provider: Provider,
+        evm_config: EvmConfig,
+        task_spawner: Box<dyn TaskSpawner>,
+    ) -> Self {
         let builder = OpPayloadBuilder::new(evm_config);
-        let inner = OpDebugWitnessApiInner { provider, builder };
+        let inner = OpDebugWitnessApiInner { provider, builder, task_spawner };
```

Comment:
good start, we also want to add a new tokio Semaphor for ratelimiting

Reply:
what would be the number of permits? 

and I'm not sure if this is the correct place and way to use it :(

```
impl<Provider, EvmConfig> OpDebugWitnessApi<Provider, EvmConfig> {
    /// Creates a new instance of the `OpDebugWitnessApi`.
    pub fn new(
        provider: Provider,
        evm_config: EvmConfig,
        task_spawner: Box<dyn TaskSpawner>,
    ) -> Self {
        let builder = OpPayloadBuilder::new(evm_config);
        let semaphore = Arc::new(Semaphore::new(3));
        let inner = OpDebugWitnessApiInner { provider, builder, task_spawner, semaphore };
        Self { inner: Arc::new(inner) }
    }
}
```

Reply:
3 seems fine